### ISSUE TITLE
Panic at tree service

### ISSUE THREAD

**Issue by alexvanin** [nspcc-dev/neofs-node#1481]
neofs-node 0.28.1-112-g4a50bdfa (`tree-service` branch).

During launch of [s3-tests](), neofs-node throws a panic and restarts at [versioned_concurrent_object_create_concurrent_remove](https://github.com/nspcc-dev/s3-tests/blob/5b19cc1bf61c0f96231dee787fde07b2f91d6368/s3tests_boto3/functional/test_s3.py#L9178) test.

```
2022-06-06T07:33:59.495Z        info    log/log.go:13   local object storage operation  {"address": "H4RR2jRsaaEZZfgYEgiFn23sUn8fLX2vyqkmA5HpWJzS/Edr8ZjWmMRZmrLFhU4NDQ1XSxU4f35LYuW4sBLT3FBaH", "op": "metabase PUT"}
2022-06-06T07:33:59.495Z        debug   delete/delete.go:37     operation finished successfully {"component": "Object.Delete service", "request": "DELETE", "address": "H4RR2jRsaaEZZfgYEgiFn23sUn8fLX2vyqkmA5HpWJzS/6RzNFKpmVr7EkgiRMS7xrZrfQNaZBEmJjVLfj5vAmcN8", "local": false, "with session": true, "with bearer": true}
panic: runtime error: index out of range [7] with length 0                                                                                                                                                            
                                                                                                                                                                                                                      
goroutine 72999 [running]:                                                                                                                                                                                            
encoding/binary.littleEndian.Uint64(...)                                                                                                                                                                              
        encoding/binary/binary.go:77                                                                                                                                                                                  
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/pilorama.(*boltForest).TreeGetByPath.func1(0xc0042d4b60)                                                                                                     
        github.com/nspcc-dev/neofs-node/pkg/local_object_storage/pilorama/boltdb.go:451 +0x5fe                                                                                                                        
go.etcd.io/bbolt.(*DB).View(0xc0034f26f0?, 0xc0012d7670)                                                                                                                                                              
        go.etcd.io/bbolt@v1.3.6/db.go:772 +0x82                                                                                                                                                                       
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/pilorama.(*boltForest).TreeGetByPath(0xc003be03d0?, {0xee, 0x9b, 0x56, 0xae, 0x24, 0xef, 0xde, 0x42, 0x26, ...}, ...)                                        
        github.com/nspcc-dev/neofs-node/pkg/local_object_storage/pilorama/boltdb.go:422 +0x12c                                                                                                                        
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*Shard).TreeGetByPath(...)                                                                                                                            
        github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard/tree.go:27                                                                                                                                     
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).TreeGetByPath(0xc0003d2500, {0xee, 0x9b, 0x56, 0xae, 0x24, 0xef, 0xde, 0x42, 0x26, ...}, ...)                                        
        github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine/tree.go:70 +0x17e                                                                                                                             
github.com/nspcc-dev/neofs-node/pkg/services/tree.(*Service).GetNodeByPath(0xc0001d5130, {0x11461e8, 0xc003b201e0}, 0xc0027f4140)                                                                                     
        github.com/nspcc-dev/neofs-node/pkg/services/tree/service.go:303 +0x2f2                                                                                                                                       
github.com/nspcc-dev/neofs-node/pkg/services/tree._TreeService_GetNodeByPath_Handler({0xf8afa0?, 0xc0001d5130}, {0x11461e8, 0xc003b201e0}, 0xc003ac6d80, 0x0)                                                         
        github.com/nspcc-dev/neofs-node/pkg/services/tree/service_grpc.pb.go:309 +0x170                                                                                                                               
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004aa000, {0x114b2a0, 0xc000188600}, 0xc00166b200, 0xc00029bc20, 0x1867ee0, 0x0)                                                                                 
        google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf                                                                                                                                                          
google.golang.org/grpc.(*Server).handleStream(0xc0004aa000, {0x114b2a0, 0xc000188600}, 0xc00166b200, 0x0)                                                                                                             
        google.golang.org/grpc@v1.41.0/server.go:1608 +0xa1b                                                                                                                                                          
google.golang.org/grpc.(*Server).serveStreams.func1.2()                                                                                                                                                               
        google.golang.org/grpc@v1.41.0/server.go:923 +0x98                                                                                                                                                            
created by google.golang.org/grpc.(*Server).serveStreams.func1                                                                                                                                                        
        google.golang.org/grpc@v1.41.0/server.go:921 +0x28a                                                                                                                                                           
```

I can go further to see what requests S3 Gateway produces, if needed.

**Comment by alexvanin** [comment]
Can't reproduce panic in `v0.28.2-45-g52183325` (experimental branch).

**Comment by fyrchik** [comment]
Closed via #1481.
